### ISSUE TITLE
packer 1.6 Incompatible API version

### ISSUE THREAD

**Issue by floh96** [lmars/packer-post-processor-vagrant-s3#61]
```
2020/07/07 15:32:59 [INFO] Packer version: 1.6.0 [go1.14.3 darwin amd64]
2020/07/07 15:32:59 [DEBUG] Discovered plugin: vagrant-s3 = /Users/fheberl/.packer.d/plugins/packer-post-processor-vagrant-s3
2020/07/07 15:32:59 using external post-processors [vagrant-s3]
2020/07/07 15:32:59 Checking 'PACKER_CONFIG' for a config file path
2020/07/07 15:32:59 'PACKER_CONFIG' not set; checking the default config file path
2020/07/07 15:32:59 Attempting to open config file: /Users/fheberl/.packerconfig
2020/07/07 15:32:59 [WARN] Config file doesn't exist: /Users/fheberl/.packerconfig
2020/07/07 15:32:59 Setting cache directory: /Users/fheberl/Projekte/packer_windows/packer_cache
2020/07/07 15:32:59 Creating plugin client for path: /usr/local/bin/packer
2020/07/07 15:32:59 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-builder-virtualbox-iso"}
2020/07/07 15:32:59 Waiting for RPC address for: /usr/local/bin/packer
2020/07/07 15:32:59 packer-builder-virtualbox-iso plugin: [INFO] Packer version: 1.6.0 [go1.14.3 darwin amd64]
2020/07/07 15:32:59 packer-builder-virtualbox-iso plugin: Checking 'PACKER_CONFIG' for a config file path
2020/07/07 15:32:59 packer-builder-virtualbox-iso plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/07/07 15:32:59 packer-builder-virtualbox-iso plugin: Attempting to open config file: /Users/fheberl/.packerconfig
2020/07/07 15:32:59 packer-builder-virtualbox-iso plugin: [WARN] Config file doesn't exist: /Users/fheberl/.packerconfig
2020/07/07 15:32:59 packer-builder-virtualbox-iso plugin: Setting cache directory: /Users/fheberl/Projekte/packer_windows/packer_cache
2020/07/07 15:32:59 packer-builder-virtualbox-iso plugin: args: []string{"packer-builder-virtualbox-iso"}
2020/07/07 15:32:59 packer-builder-virtualbox-iso plugin: Plugin address: unix /var/folders/h0/yll3kmr55r3472sx5_00rs600000gq/T/packer-plugin065998229
2020/07/07 15:32:59 packer-builder-virtualbox-iso plugin: Waiting for connection...
2020/07/07 15:32:59 Received unix RPC address for /usr/local/bin/packer: addr is /var/folders/h0/yll3kmr55r3472sx5_00rs600000gq/T/packer-plugin065998229
2020/07/07 15:32:59 packer-builder-virtualbox-iso plugin: Serving a plugin connection...
2020/07/07 15:32:59 Creating plugin client for path: /usr/local/bin/packer
2020/07/07 15:32:59 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-provisioner-file"}
2020/07/07 15:32:59 Waiting for RPC address for: /usr/local/bin/packer
2020/07/07 15:32:59 packer-provisioner-file plugin: [INFO] Packer version: 1.6.0 [go1.14.3 darwin amd64]
2020/07/07 15:32:59 packer-provisioner-file plugin: Checking 'PACKER_CONFIG' for a config file path
2020/07/07 15:32:59 packer-provisioner-file plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/07/07 15:32:59 packer-provisioner-file plugin: Attempting to open config file: /Users/fheberl/.packerconfig
2020/07/07 15:32:59 packer-provisioner-file plugin: [WARN] Config file doesn't exist: /Users/fheberl/.packerconfig
2020/07/07 15:32:59 packer-provisioner-file plugin: Setting cache directory: /Users/fheberl/Projekte/packer_windows/packer_cache
2020/07/07 15:32:59 packer-provisioner-file plugin: args: []string{"packer-provisioner-file"}
2020/07/07 15:32:59 packer-provisioner-file plugin: Plugin address: unix /var/folders/h0/yll3kmr55r3472sx5_00rs600000gq/T/packer-plugin757594545
2020/07/07 15:32:59 Received unix RPC address for /usr/local/bin/packer: addr is /var/folders/h0/yll3kmr55r3472sx5_00rs600000gq/T/packer-plugin757594545
2020/07/07 15:32:59 packer-provisioner-file plugin: Waiting for connection...
2020/07/07 15:32:59 packer-provisioner-file plugin: Serving a plugin connection...
2020/07/07 15:32:59 Creating plugin client for path: /usr/local/bin/packer
2020/07/07 15:32:59 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-provisioner-powershell"}
2020/07/07 15:32:59 Waiting for RPC address for: /usr/local/bin/packer
2020/07/07 15:32:59 packer-provisioner-powershell plugin: [INFO] Packer version: 1.6.0 [go1.14.3 darwin amd64]
2020/07/07 15:32:59 packer-provisioner-powershell plugin: Checking 'PACKER_CONFIG' for a config file path
2020/07/07 15:32:59 packer-provisioner-powershell plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/07/07 15:32:59 packer-provisioner-powershell plugin: Attempting to open config file: /Users/fheberl/.packerconfig
2020/07/07 15:32:59 packer-provisioner-powershell plugin: [WARN] Config file doesn't exist: /Users/fheberl/.packerconfig
2020/07/07 15:32:59 packer-provisioner-powershell plugin: Setting cache directory: /Users/fheberl/Projekte/packer_windows/packer_cache
2020/07/07 15:32:59 packer-provisioner-powershell plugin: args: []string{"packer-provisioner-powershell"}
2020/07/07 15:32:59 packer-provisioner-powershell plugin: Plugin address: unix /var/folders/h0/yll3kmr55r3472sx5_00rs600000gq/T/packer-plugin069117293
2020/07/07 15:32:59 packer-provisioner-powershell plugin: Waiting for connection...
2020/07/07 15:32:59 Received unix RPC address for /usr/local/bin/packer: addr is /var/folders/h0/yll3kmr55r3472sx5_00rs600000gq/T/packer-plugin069117293
2020/07/07 15:32:59 packer-provisioner-powershell plugin: Serving a plugin connection...
2020/07/07 15:32:59 Creating plugin client for path: /usr/local/bin/packer
2020/07/07 15:32:59 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-post-processor-vagrant"}
2020/07/07 15:32:59 Waiting for RPC address for: /usr/local/bin/packer
2020/07/07 15:32:59 packer-post-processor-vagrant plugin: [INFO] Packer version: 1.6.0 [go1.14.3 darwin amd64]
2020/07/07 15:32:59 packer-post-processor-vagrant plugin: Checking 'PACKER_CONFIG' for a config file path
2020/07/07 15:32:59 packer-post-processor-vagrant plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/07/07 15:32:59 packer-post-processor-vagrant plugin: Attempting to open config file: /Users/fheberl/.packerconfig
2020/07/07 15:32:59 packer-post-processor-vagrant plugin: [WARN] Config file doesn't exist: /Users/fheberl/.packerconfig
2020/07/07 15:32:59 packer-post-processor-vagrant plugin: Setting cache directory: /Users/fheberl/Projekte/packer_windows/packer_cache
2020/07/07 15:32:59 packer-post-processor-vagrant plugin: args: []string{"packer-post-processor-vagrant"}
2020/07/07 15:32:59 packer-post-processor-vagrant plugin: Plugin address: unix /var/folders/h0/yll3kmr55r3472sx5_00rs600000gq/T/packer-plugin414012441
2020/07/07 15:32:59 Received unix RPC address for /usr/local/bin/packer: addr is /var/folders/h0/yll3kmr55r3472sx5_00rs600000gq/T/packer-plugin414012441
2020/07/07 15:32:59 packer-post-processor-vagrant plugin: Waiting for connection...
2020/07/07 15:32:59 packer-post-processor-vagrant plugin: Serving a plugin connection...
2020/07/07 15:32:59 Creating plugin client for path: /Users/fheberl/.packer.d/plugins/packer-post-processor-vagrant-s3
2020/07/07 15:32:59 Starting plugin: /Users/fheberl/.packer.d/plugins/packer-post-processor-vagrant-s3 []string{"/Users/fheberl/.packer.d/plugins/packer-post-processor-vagrant-s3"}
2020/07/07 15:32:59 Waiting for RPC address for: /Users/fheberl/.packer.d/plugins/packer-post-processor-vagrant-s3
2020/07/07 15:32:59 packer-post-processor-vagrant-s3 plugin: 2020/07/07 15:32:59 Plugin minimum port: 10000
2020/07/07 15:32:59 packer-post-processor-vagrant-s3 plugin: 2020/07/07 15:32:59 Plugin maximum port: 25000
2020/07/07 15:32:59 packer-post-processor-vagrant-s3 plugin: 2020/07/07 15:32:59 Plugin address: unix /var/folders/h0/yll3kmr55r3472sx5_00rs600000gq/T/packer-plugin216502173
2020/07/07 15:32:59 packer-post-processor-vagrant-s3 plugin: 2020/07/07 15:32:59 Waiting for connection...
Error: Failed to initialize build "virtualbox-iso"


error initializing post-processor 'vagrant-s3': Incompatible API version with
plugin. Plugin version: 4, Ours: 5

2020/07/07 15:32:59 Build debug mode: true
2020/07/07 15:32:59 Force build: false
2020/07/07 15:32:59 On error: ask
2020/07/07 15:32:59 Waiting on builds to complete...
==> Builds finished but no artifacts were created.
error initializing post-processor 'vagrant-s3': Incompatible API version with
2020/07/07 15:32:59 [INFO] (telemetry) Finalizing.
plugin. Plugin version: 4, Ours: 5
```

Plugin uses API Version 4 Packer 1.6 uses API Version 5.
Could you update the plugin to support packer 1.6?

**Comment by saintzyo** [comment]
The same issue is observed. @lmars Could you update the plugin, please?

**Comment by falconws** [comment]
Any workaround?
Do I have to downgrade packer version 1.5.x?

**Comment by floh96** [comment]
I've used the shell-local post processor and the aws-cli to upload the boxes.
Snippet from packer template:
```
      {
        "type": "shell-local",
        "environment_vars": ["box_organization={{ user `box_organization` }}", "box_name={{ user `box_name` }}", "box_version={{ user `box_version` }}", "box_description={{ user `box_description` }}"],
        "script": "./scripts/upload_vagrant_box_to_s3.sh"
      }
```
Script:

```
#!/bin/bash
set -e

box_checksum=$(sha256sum ${box_name}_virtualbox.box | awk '{print $1}')
s3_box_url="s3://nww-vagrant-images/$box_name/${box_name}_virtualbox.box"
s3_json_url="s3://nww-vagrant-images/$box_name/$box_name.json"

echo Start upload for $box_organization/$box_name Version $box_version...
aws s3 mv ./$box_name\_virtualbox.box $s3_box_url
echo Create JSON file for $box_organization/$box_name Version $box_version...
cat <<EOF >> ${box_name}_metadata.json
{
  "name": "$box_organization/$box_name",
  "description": "$box_description",
  "versions": [
    {
      "version": "$box_version",
      "providers": [
        {
          "name": "virtualbox",
          "url": "$s3_box_url",
          "checksum_type": "sha256",
          "checksum": "$box_checksum"
        }
      ]
    }
  ]
}
EOF
echo Start upload for $box_name\_metadata.json...
aws s3 mv ./$box_name\_metadata.json $s3_json_url
echo Job completed successfully!
```
The script doesn't support storring multiple versions.
Aws-cli needs to be installed on the local machine.

**Comment by matt-richardson** [comment]
I've just run into this one myself with packer 1.7.2.
Looks like it may be fixed via https://github.com/lmars/packer-post-processor-vagrant-s3/pull/62 though.
Any chance that could be merged?

**Comment by floh96** [comment]
Fixed by #62 thanks @nathan-muir and @Jalle19 for merging 🎉

**Comment by floh96** [comment]
@Jalle19 @lmars could you create a new release please?
